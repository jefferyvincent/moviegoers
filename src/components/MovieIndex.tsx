import React, {useReducer, useEffect} from 'react';
import axios from 'axios';

import appReducer from '../Reducer';
import Context from '../Context';

import Layout from './Layout';
import MovieFilter from './MovieFilter';
import Grid from './Grid';
import Pager from './Pager';

const InitialState = {
    filter: "now_playing",
    page: 1,
    movies: [],
}
  
const filter_data = [
    {
      name: 'now_playing',
      title: 'Now Playing'
    },
    {
      name: 'popular',
      title: 'Popular'
    },
    {
      name: 'top_rated',
      title: 'Top Rated'
    }
];

interface MovieIndexProps {
    match: any;
    history: any;
};


const MovieIndex: React.FC<MovieIndexProps> = ({match, history}) => {
    const [state, dispatch] = useReducer(appReducer, InitialState);
    let pager = match.params.page;

    if (pager === undefined) {
        pager = 1;
    }
    
    useEffect(() => {
      if (state.filter !== '') {
          const fetchData = async () => {
            const result = await axios(
              `https://api.themoviedb.org/3/movie/${state.filter}?page=${pager}&api_key=438581175542f3ba5208b33d46f87151`,
            );

            dispatch({
              type: 'updateMovies',
              payload: result.data
            });

            dispatch({
                type: 'updatePage',
                payload: pager
            });
          };
          fetchData();
      }
    }, [state.filter, pager]);

    //console.log(state);

    return (
      <Context.Provider value={dispatch}>
        <Layout>
          <h2>Ultimate Movie Guide</h2>
          <MovieFilter filters={filter_data} active_filter={state.filter} history={history} />
          <Grid movies={state.movies} />
          <Pager page={state.page} max={state.movies.total_pages} history={history} />
        </Layout>
      </Context.Provider>
    );
}

export default MovieIndex;