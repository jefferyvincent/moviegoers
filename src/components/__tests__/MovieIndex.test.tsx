import React from 'react';
import { shallow } from 'enzyme';

import MovieIndex from '../MovieIndex';

describe('<MovieIndex/>', () => {
    it('renders without crashing', () => {
        shallow(<MovieIndex match={{params:{page: 1}}} history={[]} />);
    });
});

