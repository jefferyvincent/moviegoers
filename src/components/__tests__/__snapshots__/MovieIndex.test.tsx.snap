// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieIndex/> renders 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <header
      class="app-header"
    >
      <div
        class="header-left"
      >
        <a
          href="#/"
          title="Ultimate Moviegoers Guide"
        >
          <h1
            class="header-logo"
          >
            <img
              alt="Ultimate Moviegoers Guide"
              src="logo.png"
            />
          </h1>
        </a>
      </div>
      <div
        class="header-right"
      >
        <div
          class="app-search"
        >
          <input
            autocomplete="off"
            autocorrect="off"
            placeholder="search it."
            spellcheck="false"
          />
          <img
            alt="Search"
            class="search-mag"
            src="mag.svg"
          />
        </div>
      </div>
    </header>
    <main
      class="content"
    >
      <h2>
        Ultimate Movie Guide
      </h2>
      <div
        class="movieFilter"
      >
        <h3
          class="filter-text"
        >
          Showing Movies sorted by Now Playing
        </h3>
        <div
          class="filter"
        >
          <div
            class="filter-selected"
          >
            Now Playing 
            <img
              alt="filter"
              src="arrow.svg"
            />
          </div>
          <ul
            class="filter-options hidden"
          >
            <li>
               Popular
            </li>
            <li>
               Top Rated
            </li>
          </ul>
        </div>
      </div>
      <ul
        class="grid"
      >
        <li>
          <div
            class="loader"
          >
            Loading...
          </div>
        </li>
      </ul>
    </main>
    <footer
      class="footer"
    >
      <hr />
      <div
        class="copyright"
      >
        Â© 2019 Ultimate Moviegoers Guide
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
